src/Pages/Cartpage/CartProduct.jsx

```diff
@@ -0,0 +1,178 @@
+import { useState, useEffect } from "react";
+import { useCart, useAuth, useAlert, useWishlist } from "../../CustomHooks";
+import { Card } from "../../Components/UI";
+import { FaStar } from "react-icons/fa";
+import "./CartProduct.css";
+export const CartProduct = ({ product }) => {
+  const { title, subtitle, price, oldPrice, image, discount, rating } = product;
+  const [isDisabled, setIsDisabled] = useState(false);
+  const { cartState, updateCart, removeFromCart, error } = useCart();
+  const { showAlert, setShowAlert } = useAlert();
+  const { authState } = useAuth();
+  const { wishlistState, addToWishlist } = useWishlist();
+  const existsInWishlist = wishlistState.items.some(
+    (item) => item._id === product._id
+  );
+
+  useEffect(() => {
+    setTimeout(() => {
+      setIsDisabled(false);
+    }, 1000);
+  }, [isDisabled]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setShowAlert({
+        showAlert: false,
+        alertMessage: null,
+        type: null,
+      });
+    }, 2000);
+  }, [showAlert.showAlert]);
+
+  const qtyIncrementHandler = () => {
+    updateCart({
+      product,
+      type: "increment",
+      token: authState.token,
+    });
+    setIsDisabled(true);
+    if (error) {
+      setShowAlert({
+        showAlert: true,
+        alertMessage: error,
+        type: "danger",
+      });
+    } else {
+      setShowAlert({
+        showAlert: true,
+        alertMessage: "Quantity Increased Successfully!",
+        type: "success",
+      });
+    }
```

Comment:
Here, I believe you're trying to show to different alerts based on if there **error** or not. The rest code(calling the setShowAlert method based on the condition) is repeatitive. Always follow DRY approach(DONT REPEAT YOURSELF)
What you can do here:
```
// Other code
 setIsDisabled(true);
const alertMessage =  error || "Quantity Increased Successfully!",
const type = error ? "danger":"success" 
setShowAlert({
 showAlert: true,
 alertMessage,
 type,
})
```